extensions/amp-geo/0.1/amp-geo.js

```diff
@@ -256,6 +299,16 @@ export class AmpGeo extends AMP.BaseElement {
         });
   }
 
+  /**
+   * Ensure that subdivisions allowed for a specific country
+   * @param {string} country
+   * @return {boolean}
+   * @private
+   */
+  areSubdivisionsAllowed_(country) {
```

Comment:
I'm not sure that we need to hardcode this. The AMP Cache only currently provides `us-ca`, but if you're using your own geo provider we should be able to support anything they return. It would also mean less code being shipped here. 😄

Reply:
I thought I would support all subdivisions, but I found out that there could be a subdivision with 3 symbol lengths (Russian subdivision) (maybe it's the max length for subdivisions in iso) and I thought if we support all subdivisions it could make code more fragile and should check all subdivisions for sure )

Reply:
Oo, that's a good point. Actually, some subdivisions are numbers instead of letter, and can have 1-3 chars total. I think updating to match `[0-9a-zA-z]{1,3}` would be fine.

Reply:
I made some changes with regexps for the future) Please check

Reply:
LGTM, let's remove `areSubdivisionsAllowed_` and `countriesWithSubdivision`

Reply:
Maybe in the next PR, I think need a small investigation with the subdivision iso? Now we could check that this changes work with current restrictions

Reply:
Please check, I removed restrictions)

Reply:
So, I couldn't get info about subdivisions format from ISO site, but got info from geo IP site(maybe not correct), they use string(1-3) format or subdivisions, I hope this enough for the investigation))